tests/parametric/test_headers_tracecontext.py

```diff
@@ -807,6 +807,165 @@ def test_tracestate_w3c_p_inject(self, test_agent, test_library):
+            with test_library.start_span(
+                name="matching_trace_id_and_span_id_missing_p",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000002-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2,foo=1"],
+                    ["x-datadog-trace-id", "2"],
+                    ["x-datadog-parent-id", "987654321"],
+                ],
+            ):
+                pass
+
+            # 3) Datadog and tracecontext headers, only trace ids matches
+            with test_library.start_span(
+                name="matching_trace_id_only",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000003-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-trace-id", "3"],
+                    ["x-datadog-parent-id", "9876543210"],
+                ],
+            ):
+                pass
+
+            # 4) Datadog headers only
+            with test_library.start_span(
+                name="only_datadog_headers",
+                http_headers=[["x-datadog-trace-id", "4"], ["x-datadog-parent-id", "987654321"],],
+            ):
+                pass
+
+            # 5) Datadog and tracecontext headers, no x-datadog-trace-id, only tid DD tag matches
+            with test_library.start_span(
+                name="only_tid_and_tracecontext_trace_id_match",
+                http_headers=[
+                    ["traceparent", "00-00000000000000000000000000000005-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-parent-id", "9876543210"],
+                    ["x-datadog-trace-id", "8"],
+                    ["x-datadog-tags", "_dd.p.tid=5"],
+                ],
+            ):
+                pass
+
+            # 6) Datadog and tracecontext headers, not x-datadog-trace-id, tid and the parent ids match
+            with test_library.start_span(
+                name="matching_tid_and_span_id",
+                http_headers=[
+                    ["traceparent", "00-00000000000000000000000000000006-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-parent-id", "987654321"],
+                    ["x-datadog-tags", "_dd.p.tid=6"],
+                ],
+            ):
+                pass
+
+        traces = test_agent.wait_for_num_traces(6)
+
+        assert len(traces) == 6
+        case1, case2, case3, case4, case5, case6 = (
+            traces[0][0],
+            traces[1][0],
+            traces[2][0],
+            traces[3][0],
+            traces[4][0],
+            traces[5][0],
+        )
+
+        # 1) Datadog and tracecontext headers, trace-id and span-id match
+        assert case1["name"] == "matching_trace_id_and_span_id"
+        assert case1["meta"]["_dd.parent_id"] == "0123456789abcdef"
```

Comment:
I'm confused here, has the RFC changed with validation between 128-bit Trace IDs and 64-bit Trace IDs?

This test had the following input:

```
                    ["traceparent", "00-11111111111111110000000000000001-000000003ade68b1-01"],
                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
                    ["x-datadog-trace-id", "1"],
                    ["x-datadog-parent-id", "987654321"],
```

The trace IDs don't match, so there should be no `_dd.parent_id` here